### ISSUE TITLE
关于main_program和startup_program

### ISSUE THREAD

**Issue by zhangxuanaj** [PaddlePaddle/models#4907]
PaddleNLP/emotion_detection/run_classifier.py
178-184行：

        with fluid.program_guard(train_program, startup_prog):
            with fluid.unique_name.guard():
                train_loader, loss, accuracy, num_seqs = create_model(
                    args, num_labels=num_labels, is_prediction=False)

                sgd_optimizer = fluid.optimizer.Adagrad(learning_rate=args.lr)
                sgd_optimizer.minimize(loss)

请问下，训练完成后，startup_prog是不是拥有和train_program一样的参数？

**Comment by Aurelius84** [comment]
> startup_prog是不是拥有和train_program一样的参数？

它们指向同一份参数，所以是一样的。`startup_prog`是用来在训练开始阶段，通过调用`exe.run(startup_prog)`来进行参数初始化的。只关注train_program即可
